compiler-plugin/src/main/java/io/ballerina/lib/np/compilerplugin/PromptAsCodeCodeModificationTask.java

```diff
@@ -214,6 +226,33 @@ public ImportDeclarationNode transform(ImportDeclarationNode importDeclarationNo
         }
     }
 
+    private static class FunctionCallModifier extends TreeModifier {
+        private final ModifierData modifierData;
+        private final SourceModifierContext modifierContext;
+        private final ModuleId moduleId;
+        private final Document document;
+
+        FunctionCallModifier(ModifierData modifierData, SourceModifierContext modifierContext, ModuleId moduleId,
+                             Document document) {
+            this.modifierData = modifierData;
+            this.modifierContext = modifierContext;
+            this.moduleId = moduleId;
+            this.document = document;
+        }
+
+        @Override
+        public FunctionCallExpressionNode transform(FunctionCallExpressionNode functionCallExpressionNode) {
```

Comment:
Why didn't we do this only for np:callLlm calls?

Reply:
You mean to special case only for `np:callLlm`, right? There's no issue with the current implementation, right?it's to analyze all function calls.

Reply:
https://github.com/ballerina-platform/ballerina-library/issues/7746#issuecomment-2768296136